### ISSUE TITLE
Infoboxer::Parser::ParsingError on wiki table cells.

### ISSUE THREAD

**Issue by phlegx** [molybdenum-99/infoboxer#66]
Hi! I tried out your gem and if I do:

```ruby
Infoboxer.wikipedia.get('Area')
```

I get:

```log
Infoboxer::Parser::ParsingError: Infinite loop on position 21 at line 242:
        }}{4}s^2\,\!</math>
        from /lib/infoboxer/parser/context.rb:145:in `fail!'
        from /lib/infoboxer/parser/util.rb:88:in `block in guarded_loop'
        from /lib/infoboxer/parser/util.rb:83:in `loop'
        from /gems/infoboxer-0.2.4/lib/infoboxer/parser/util.rb:83:in `guarded_loop'
        from /gems/infoboxer-0.2.4/lib/infoboxer/parser/table.rb:80:in `table_cells'
        from /gems/infoboxer-0.2.4/lib/infoboxer/parser/table.rb:49:in `table_next_line'
        from /gems/infoboxer-0.2.4/lib/infoboxer/parser/table.rb:18:in `block in table'
        from /gems/infoboxer-0.2.4/lib/infoboxer/parser/table.rb:17:in `loop'
        from /gems/infoboxer-0.2.4/lib/infoboxer/parser/table.rb:17:in `table'
        from /gems/infoboxer-0.2.4/lib/infoboxer/parser/paragraphs.rb:26:in `paragraph'
        from /gems/infoboxer-0.2.4/lib/infoboxer/parser/paragraphs.rb:10:in `paragraphs'
       ...
```

This is the line that makes problems: `||<math>\frac{\sqrt{3}}{4}s^2\,\!</math>`
Source: https://en.wikipedia.org/w/index.php?title=Area&action=edit

Wiki code snipped:

```
|Regular [[triangle]] ([[equilateral triangle]])
||<math>\frac{\sqrt{3}}{4}s^2\,\!</math>
||<math>s</math> is the length of one side of the triangle.
|-
```

**What can I do to solve this problem?**

**Comment by zverok** [comment]
OK, seems Infoboxer doesn't play well currently with complicated math (especially inside tables). Looking into it.

**Comment by phlegx** [comment]
@zverok you are very fast to reply! :+1:  Thank you! First, I need to understand the code of the gem, then I can help you. :smile:

**Comment by phlegx** [comment]
This is the line that makes problems: `||<math>\frac{\sqrt{3}}{4}s^2\,\!</math>`
Source: https://en.wikipedia.org/w/index.php?title=Area&action=edit

**Comment by zverok** [comment]
Yep, exactly. Should be fixed now in `develop` branch.

```ruby
page = Infoboxer.wikipedia.get('Area')
# => #<Page(title: "Area", url: "https://en.wikipedia.org/wiki/Area"): Area is the quantity that expre...> 
page.tables.first.rows.each { |r| puts r.cells[1] }
# Formula
# |\frac{\sqrt{3}}{4}s^2\,\!
# \sqrt{s(s-a)(s-b)(s-c)}\,\!
# \tfrac12 a b \sin(C)\,\!
# \tfrac12bh \,\!
# \frac{1}{2}b\sqrt{a^2-\frac{b^2}{4}}=\frac{b}{4}\sqrt{4a^2-b^2}
# \tfrac12ab
# bh\,\!
# \frac{(a+b)h}{2} \,\!
# \frac{3}{2} \sqrt{3}s^2\,\!
# 2(1+\sqrt{2})s^2\,\!
# \frac{1}{4}nl^2\cdot \cot(\pi/n)\,\!
# \frac{1}{4n}p^2\cdot \cot(\pi/n)\,\!
# \frac{1}{2}nR^2\cdot \sin(2\pi/n) = nr^2 \tan(\pi/n)\,\!
# \tfrac12 ap = \tfrac12 nsa \,\!
# \pi r^2\ \text{or}\ \frac{\pi d^2}{4} \,\!
# \frac{\theta}{2}r^2\ \text{or}\ \frac{L \cdot r}{2}\,\!
# \pi ab \,\!
# 2\pi r (r + h)\,\!
# 2 \pi r h \,\!
# 4\pi r^2\ \text{or}\ \pi d^2\,\!
# B+\frac{P L}{2}\,\!
# B+\frac{P L}{2}\,\!
# \frac{4}{\pi} A\,\!
# \frac{\pi}{4} C\,\!
```

**Comment by phlegx** [comment]
:smile: cooooool. Thank you! I try it out.

**Comment by phlegx** [comment]
It works! Nice work @zverok . So, I continue to play with your gem :smiley: 
When it is included in the next release?

**Comment by zverok** [comment]
The development is a bit stale now, much of other work.
So, for the sake of library, I've just released 0.2.5 with this fix. You can now install it from RubyGems.

**Comment by zverok** [comment]
Sorry, somehow managed to release empty (= no changes) version. 0.2.6 should suit.
